### PR TITLE
Added Support for Dark Mode  🌙

### PR DESCRIPTION
**Pull request by MathurAditya724**
Issue - #1006 

- [x] Added CSS Variables for managing the colors
- [x] Changes mode based on the user's color preference
- [x] Added spacing between the sponsor text and the content 😮‍💨

I have changed the color scheme according to my taste. Feel free to change it or let me know if any more changes are required.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -0,0 +1,771 @@
+lockfileVersion: '6.0'
```
Comment (styfle) [inline pnpm-lock.yaml:1]: Please remove this pnpm lock file

This repo uses yarn as seen here https://github.com/styfle/packagephobia/blob/main/yarn.lock

File: src/components/Sponsors.tsx [hunk 2]
```diff
@@ -1,106 +1,107 @@
 import React from 'react';
 
+const SPONSORS = [
```
Comment (styfle) [inline src/components/Sponsors.tsx:3]: Can you explain the change here? 

Is this relate to dark mode?

File: src/components/Stats.tsx [hunk 3]
```diff
@@ -24,7 +24,14 @@ type StatProp = {
...
 export function Stat(props: StatProp) {
-    const { size, unit, label, scale = 1, color = '#212121', textAlign = 'center' } = props;
+    const {
+        size,
+        unit,
+        label,
+        scale = 1,
+        color = 'var(--foreground)',
+        textAlign = 'center',
+    } = props;
```
Comment (styfle) [inline src/components/Stats.tsx:34]: Similarly here, is this change necessary?

File: src/pages/compare.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ export default ({ inputStr, results }: CompareProps) => {
             <PageContainer>
                 <SearchBar autoFocus={false} defaultValue={inputStr} />
                 <div style={{ maxWidth: '100%', overflow: 'auto' }}>
-                    <table style={{ marginTop: '30px' }}>
+                    <table className="compare" style={{ marginTop: '30px' }}>
```
Comment (styfle) [inline src/pages/compare.tsx:34]:
```suggestion
                    <table style={{ marginTop: '30px' }}>
```

Looks like this className is unused so it can be removed

File: src/components/SearchBar.tsx [hunk 5]
```diff
@@ -12,14 +12,15 @@ const height = '4rem';
 const radius = '3px';
 
 const input: React.CSSProperties = {
-    color: '#333',
+    color: 'var(--muted-foreground)',
+    background: 'var(--muted)',
```
Comment (styfle) [inline src/components/SearchBar.tsx:16]: This background looks too dark in light mode

- [Before](https://github.com/styfle/packagephobia/assets/229881/aef16a39-5de8-4870-86b2-7e86eef2d964")
- [After](https://github.com/styfle/packagephobia/assets/229881/a12f9ef4-b84e-47b6-8aef-efc1dcdab5a1")

Comment (styfle) [reply src/components/SearchBar.tsx:16]: Maybe we can change it to use something like this design:

https://vercel.com/design/input#prefix-and-suffix

**Comment by MathurAditya724** [comment]
I have made all the requested changes, please have a look

File: src/components/Sponsors.tsx (hunk 2 above)
Comment (MathurAditya724) [reply src/components/Sponsors.tsx]: This basically adds a bit of spacing below the Sponsors keyword, and I have made this in an array to implement DRY. Just a small change that I thought will look good.

No, it is not related to Dark Mode. I will make a separate PR for this later.

**Comment by styfle** [comment]
@MathurAditya724 I'm still seeing the dark gray input instead of white

image

https://packagephobia-git-fork-mathuraditya724-main-packagephobia.vercel.app/result?p=satori

Can you ensure that the light mode colors are not changed from the original here:

https://packagephobia.com/result?p=satori

**Comment by styfle** [comment]
@MathurAditya724 The search icon is nearly invisible 

https://packagephobia-git-fork-mathuraditya724-main-packagephobia.vercel.app/

image

**Comment by MathurAditya724** [comment]
All good?

**Comment by styfle** [comment]
@MathurAditya724 Light mode looks good now 👍 

image

--------

However, dark mode has the colors flipped in the bar chart.

image

Can we keep the darker shade of green on the bottom?

**Comment by MathurAditya724** [comment]
This commit should resolve this issue

File: src/components/BarGraph.tsx [hunk 6]
```diff
@@ -29,13 +29,13 @@ const stylesheet = `
     cursor:pointer;
 }
 .color1 {
-    background: #32DE85;
+    background: #32de85;
```
Comment (styfle) [inline src/components/BarGraph.tsx:32]:
```suggestion
    background: #32DE85;
```

**Review by styfle — APPROVED** [review]
Great, thanks! 🎉

**Comment by XhmikosR** [comment]
@styfle just wondering, wouldn't it make more sense if the new CSS file was inlined too?

**Comment by styfle** [comment]
I'm starting to think inlining is not the right solution since it can't be cached between page navigations.

**Comment by XhmikosR** [comment]
That makes sense too, as long as you move all CSS to an external site.

That being said, you need cache busting, and preload if you don't want to use a blocking CSS. Your call, I just noticed the inconsistency and it's easier to just inline the newly added selectors like the rest.
